.devops/templates/cleanup.yml

```diff
@@ -2,13 +2,36 @@ parameters:
   - name: checkForChangedFiles
     type: boolean
     default: true
+  - name: runComponentGovernance
+    type: boolean
+    default: false
 
 steps:
   - script: |
       npm run check-for-changed-files
     condition: ${{ parameters.checkForChangedFiles }}
     displayName: check for changed files
 
+  # The cleanup task breaks the component governance task which runs automatically on master builds
+  # (via a pipeline decorator from an extension)...so run that here before cleanup if appropriate.
+  # - ${{ if eq(variables['Build.SourceBranch'], 'refs/heads/master') }}:
+  - ${{ if eq(parameters.runComponentGovernance, true) }}:
+      - task: ComponentGovernanceComponentDetection@0
+        displayName: 'Component Detection'
+        inputs:
+          sourceScanPath: $(Agent.BuildDirectory)
+        condition: succeeded()
+        timeoutInMinutes: 5
+        continueOnError: true
+
+  # If it's a pipeline where the decorator would generally run the governance task, but it hasn't
+  # been requested for this template instance, set a variable to skip it.
+  # (For example, CI builds only need to run this on one of the jobs, not every job.)
+  # prettier-ignore
+  - ${{ if and(eq(parameters.runComponentGovernance, true), eq(variables['Build.SourceBranch'], 'refs/heads/master')) }}:
+    - script: echo "##vso[task.setvariable variable=skipComponentGovernanceDetection;]true"
```

Comment:
if we can disable, why not just setup another job that runs component governance ?

Reply:
...yeah that would probably make sense 😆

Reply:
Dang ~~boring~~ simple ways to do things, now nobody gets to see my fancy yaml syntax :P

Reply:
Updated the tools/setup template to disable the job by default, and added a step to `azure-pipelines.release.yml` to run component governance detection manually. As far as I can tell this should be adequate since the point of the task is to check installed packages, and we install the same packages in all our pipelines in each branch.